sklearn/linear_model/base.py

```diff
@@ -121,17 +121,17 @@ def _preprocess_data(X, y, fit_intercept, normalize=False, copy=True,
         sample_weight = None
 
     if check_input:
-        X = check_array(X, copy=copy, accept_sparse=['csr', 'csc'],
+        X = check_array(X, accept_sparse=['csr', 'csc'],
                         dtype=FLOAT_DTYPES)
-    elif copy:
-        if sp.issparse(X):
-            X = X.copy()
-        else:
-            X = X.copy(order='K')
 
     y = np.asarray(y, dtype=X.dtype)
 
     if fit_intercept:
+        if copy is None or copy:
```

Comment:
It seems that now `copy=True` will still not copy if fit_intercept is False?

Reply:
That's the goal yes, as there is no reason to make a copy here. As you mentioned earlier (I think), this parameter should have been called `inplace` not `copy`.

Reply:
so True behaves as None? If that's the case then maybe we could just not introduce None and change the behaviour of True instead (explaining that True means "only when necessary")?